src/Frontend/test/specs/failedmessages/edit-and-retry.spec.ts

```diff
@@ -5,6 +5,32 @@ import { getEditAndRetryEditor } from "./questions/getEditAndRetryEditor";
 import { expect } from "vitest";
 
 describe("FEATURE: Editing failed messages", () => {
+  function getBoundingClientRect(): DOMRect {
+    const rec = {
+      x: 0,
+      y: 0,
+      bottom: 0,
+      height: 0,
+      left: 0,
+      right: 0,
+      top: 0,
+      width: 0,
+    };
+    return { ...rec, toJSON: () => rec };
+  }
+
+  class FakeDOMRectList extends Array<DOMRect> implements DOMRectList {
+    item(index: number): DOMRect | null {
+      return this[index];
+    }
+  }
+
+  document.elementFromPoint = (): null => null;
+  HTMLElement.prototype.getBoundingClientRect = getBoundingClientRect;
+  HTMLElement.prototype.getClientRects = (): DOMRectList => new FakeDOMRectList();
+  Range.prototype.getBoundingClientRect = getBoundingClientRect;
+  Range.prototype.getClientRects = (): DOMRectList => new FakeDOMRectList();
```

Comment:
Ran into https://github.com/jsdom/jsdom/issues/3729#issue-2335597398 so I used that fix mentioned there